opencti-platform/opencti-front/src/private/components/analyses/reports/ReportDetails.jsx

```diff
@@ -104,23 +93,64 @@ const styles = (theme) => ({
   relatedContainers: {
     paddingTop: 0,
   },
-});
+}));
+
+const ReportDetailsFragment = graphql`
+    fragment ReportDetails_report on Report {
+        id
+        published
+        report_types
+        description
+        relatedContainers(
+            first: 10
+            orderBy: published
+            orderMode: desc
+            types: ["Report"]
+            viaTypes: ["Indicator", "Stix-Cyber-Observable"]
+        ) {
+            edges {
+                node {
+                    id
+                    entity_type
+                    ... on Report {
+                        name
+                        description
+                        published
+                        createdBy {
+                            ... on Identity {
+                                id
+                                name
+                                entity_type
+                            }
+                        }
+                        objectMarking {
+                            id
+                            definition_type
+                            definition
+                            x_opencti_order
+                            x_opencti_color
+                        }
+                    }
+                }
+            }
+        }
+    }
+`;
 
-const ReportDetailsComponent = (props) => {
-  const { t, fldt, fsd, classes, report } = props;
+const ReportDetails = ({ report }) => {
+  const classes = useStyles();
+  const { t_i18n, fldt, fsd } = useFormatter();
   const [expanded, setExpanded] = useState(false);
   const [height, setHeight] = useState(0);
   const ref = useRef(null);
+  const reportData = useFragment(ReportDetailsFragment, report);
   useEffect(() => {
     setHeight(ref.current.clientHeight);
   });
-  const expandable = report.relatedContainers.edges.length > 5;
-  const relatedContainers = R.take(
-    expanded ? 200 : 5,
-    report.relatedContainers.edges,
-  ).filter(
-    (relatedContainerEdge) => relatedContainerEdge.node.id !== report.id,
-  );
+  const expandable = reportData.relatedContainers.edges.length > 5;
+  const relatedContainers = reportData.relatedContainers.edges
+    .slice(0, expanded ? 200 : 5)
+    .filter((relatedContainerEdge) => relatedContainerEdge.node.id !== reportData.id);
```

Comment:
```suggestion
  const relatedContainers = reportData.relatedContainers.edges
    .filter((relatedContainerEdge) => relatedContainerEdge.node.id !== reportData.id)
    .slice(0, expanded ? 200 : 5);
```

If we slice then filter, we might get 4 items instead of 5 if the current report is in the first 5 items.

Reply:
indeed! fixed